main.go

```diff
@@ -520,6 +520,10 @@ func main() {
 
 			r.Use(authentication.WithTenant)
 			r.Use(authentication.WithTenantID(tenantIDs))
+			// We want to enforce that all remote write requests have an access token present in the Authorization
+			// header to avoid repeatedly redirecting clients that might be unaware of OIDC flow to the identity
+			// provider, possibly overloading it.
+			r.Use(authentication.EnforceAccessTokenPresent("/api/v1/receive"))
```

Comment:
I'm wondering if we could make this somehow more universal, since right now this will only apply to metrics / remote write, but similar situation could occur with logs and traces as well right?

Reply:
@matej-g I don't know much how writing logs/traces work, but I suspect they are susceptible to the same kind of situation. I'm happy to use this approach for these other telemetry signals in follow up PRs. :)

Reply:
We know the write endpoints upfront right? So perhaps renaming the function to something like `EnsureAccessTokenPresentOnWrite()` or similar and have a check against all endpoints?